### DISCUSSION TITLE
Add possibility to show ALL chart description by default in Superset dashboard

### DISCUSSION THREAD

**Discussion by andreabrumana** [apache/superset#22724] [Ideas]
**Background**
Chart description are very useful in order to inform stakeholders about the data shown in charts. Superset allows the possibility of inserting a description, that is not shown by default. In order to show it, one can:
- Click manually, each time, on "Toggle description" (as a dashboard user)
- Modify the .json metadata file and set `expanded_slices` as `true`, for the specified `chart_ids` (as a dashboard developer).

**Feature request**
Since may happen that we have a lot of charts and that new charts can be added to the dashboard, there should be a a possibility to set `expanded_slice`: `true` for ALL charts present in the dashboard, without the need of making it explicit this chart by chart.

**Comment by sfirke** [comment]
This is great idea for a feature.  I'd also like to see, for when someone is specifying individual charts, if there's a way to just list slice IDs instead of typing the `true` over and over like `"935": true,`

**Comment by Manish-2022** [comment]
Where can i find this .json metadata file

**Reply by abagh0703** [reply]
> Where can i find this .json metadata file

Edit dashboard --> Edit properties --> Advanced --> JSON Metadata.

You add an entry like `"expanded_slices": {"123": true},` to the root of the json. To get the value for 123, you get the slice id, which you can find in the url when editing a chart.

**Comment by sfirke** [comment]
Now I wonder if we could go one farther and set a Superset-wide config variable that would apply across dashboards to default showing a chart description to true.

**Comment by SBIN2010** [comment]
How about having an information icon displayed on the chart by default—showing a tooltip with the description on hover—while hiding the icon if the description display is enabled?
@sfirke @andreabrumana 
What do you think of that approach?

**Reply by sfirke** [reply]
@SBIN2010 I think what you propose is an improvement for sure. The two states for a description would be either shown or tool tip, with tooltip replacing invisible (the current state if not shown).

That doesn't entirely address this discussion thread, though it would make it less urgent to address.

**Reply by SBIN2010** [reply]
While looking into this, I found that the "expand_all_slices" functionality had already been added to the dashboard configuration

**Reply by SBIN2010** [reply]
implemented in https://github.com/apache/superset/pull/32958

**Reply by sfirke** [reply]
Oh! Nice find.

**Comment by SBIN2010** [comment]
For example, like this:
<img width="886" height="582" alt="Снимок экрана от 2026-08-18 00-10-04" src="https://github.com/user-attachments/assets/afa6176a-8014-422a-b417-05844619e18d" />

**Reply by SBIN2010** [reply]
PR created https://github.com/apache/superset/pull/43271

**Comment by SBIN2010** [comment]
Done in  https://github.com/apache/superset/pull/32958 and https://github.com/apache/superset/pull/43271
